proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -17298,8 +17269,10 @@ lemma send_ipc_ct_in_state:
    \<lbrace>\<lambda>_. ct_in_state P :: 'state_ext state \<Rightarrow> _\<rbrace>"
   supply if_split [split del]
   unfolding send_ipc_def
-  apply (wpsimp wp: sts_ctis_neq maybeM_inv)
-            apply (wpsimp wp: hoare_vcg_imp_lift' hoare_vcg_all_lift get_tcb_obj_ref_wp)
+  apply wpsimp
+       apply ((wpsimp wp: sts_ctis_neq)+)[3]
+    apply wpsimp
+            apply (wpsimp wp: sts_ctis_neq)
```

Comment:
That looks a bit unpleasant. What does the middle wpsimp use instead of `sts_ctis_neq`?

Reply:
It was there to break the wpsimp loop at a specific point, but a little investigation has revealed a simpler proof :) Thanks for pointing it out!